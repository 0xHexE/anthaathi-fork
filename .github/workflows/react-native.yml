name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
