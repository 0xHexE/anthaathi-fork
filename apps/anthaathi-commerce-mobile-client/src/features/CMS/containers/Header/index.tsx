import React from 'react';
import {Image, StatusBar, StyleSheet, View} from 'react-native';
import {Appbar, useTheme} from 'react-native-paper';
import {CoreComponentType} from '../../types/common';

export interface HeaderProps {
  rightIcon?: string;
  rightOnPress?: () => void;
  leftIcon?: string;
  leftOnPress?: () => void;
  title?: string;
  placement?: 'left' | 'center';
  logoImage?: string;
}

const Header = ({
  rightIcon,
  rightOnPress,
  leftIcon,
  leftOnPress,
  title = '',
  placement = 'left',
  logoImage,
}: HeaderProps) => {
  const theme = useTheme();
  return (
    <View testID="header">
      <StatusBar
        animated={true}
        backgroundColor={theme.colors.primary}
        barStyle={theme.dark ? 'dark-content' : 'light-content'}
      />
      <Appbar.Header
        style={{
          backgroundColor: theme.colors.background,
        }}>
        {leftIcon && (
          <Appbar.Action
            testID="leftOnHandler"
            icon={leftIcon}
            color="#364A15"
            onPress={leftOnPress}
          />
        )}
        {title === '' ? (
          <Image
            testID="headerImage"
            style={[
              {
                height: 48,
                width: 70,
              },
              placement === 'center' && styles.headerAlign,
            ]}
            source={{
              uri: logoImage,
            }}
          />
        ) : null}
        <Appbar.Content
          title={title}
          style={[placement === 'center' && styles.headerAlign]}
        />
        {rightIcon && (
          <Appbar.Action
            testID="rightOnHandler"
            icon={rightIcon}
            color="#364A15"
            onPress={rightOnPress}
          />
        )}
      </Appbar.Header>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  headerAlign: {
    alignItems: 'center',
  },
});

export const HeaderCMSInput = {
  _component: CoreComponentType.Header,
  component: Header,
};
