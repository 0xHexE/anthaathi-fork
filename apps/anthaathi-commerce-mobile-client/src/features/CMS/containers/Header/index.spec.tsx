import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {IntlProvider} from 'react-intl';
import locale from '../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';
import Header from './index';

describe('header', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <Header leftIcon="menu" title="Header title" />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('header')).toBeTruthy();
  });

  it('should call when we call tap leftOnPress', function () {
    const leftonpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <Header
            leftIcon="menu"
            leftOnPress={leftonpress}
            title="Header title"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('leftOnHandler')!);
    expect(leftonpress).toBeCalledTimes(1);
  });

  it('should call when we call tap rightOnPress', function () {
    const rightonpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <Header
            rightOnPress={rightonpress}
            title="Header title"
            rightIcon="menu"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('rightOnHandler')!);
    expect(rightonpress).toBeCalledTimes(1);
  });

  it('should have Header title', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <Header title="Header title" />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Header title')).toBeTruthy();
  });

  it('should have Header Image', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <Header logoImage="https://cdn.shopify.com/s/files/1/0648/1303/9842/files/Newsletter_bg_300x.png?v=1653648705" />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByTestId('headerImage')!.props.source).toMatchObject({
      uri: 'https://cdn.shopify.com/s/files/1/0648/1303/9842/files/Newsletter_bg_300x.png?v=1653648705',
    });
  });
});
