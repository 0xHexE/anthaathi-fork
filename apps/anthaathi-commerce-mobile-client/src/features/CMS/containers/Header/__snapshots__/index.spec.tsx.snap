// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should render the item 1`] = `
<View
  testID="header"
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 251, 254, 1)",
          "elevation": 0,
          "paddingTop": 0,
          "zIndex": 0,
        },
        Object {},
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(255, 251, 254, 1)",
              "flexDirection": "row",
              "height": 64,
              "paddingHorizontal": 4,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "flexDirection": "column",
                  "paddingTop": 8,
                },
                Object {
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0,
                        "shadowRadius": 0,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        Object {
                          "backgroundColor": undefined,
                          "borderColor": "rgba(121, 116, 126, 1)",
                          "borderRadius": 20,
                          "borderWidth": 0,
                          "elevation": 0,
                          "height": 40,
                          "margin": 6,
                          "overflow": "hidden",
                          "width": 40,
                        }
                      }
                    >
                      <View
                        accessibilityRole="button"
                        accessibilityState={
                          Object {
                            "disabled": true,
                          }
                        }
                        accessible={true}
                        focusable={false}
                        hitSlop={
                          Object {
                            "bottom": 6,
                            "left": 6,
                            "right": 6,
                            "top": 6,
                          }
                        }
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Array [
                            Object {
                              "overflow": "hidden",
                            },
                            Object {
                              "alignItems": "center",
                              "flexGrow": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                        testID="leftOnHandler"
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "justifyContent": "center",
                              },
                              Object {
                                "height": 24,
                                "width": 24,
                              },
                            ]
                          }
                        >
                          <View
                            collapsable={false}
                            style={
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "opacity": 1,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                                "transform": Array [
                                  Object {
                                    "rotate": "0deg",
                                  },
                                ],
                              }
                            }
                          >
                            <Text
                              accessibilityElementsHidden={true}
                              allowFontScaling={false}
                              importantForAccessibility="no-hide-descendants"
                              pointerEvents="none"
                              selectable={false}
                              style={
                                Array [
                                  Object {
                                    "color": "rgba(73, 69, 79, 1)",
                                    "fontSize": 24,
                                  },
                                  Array [
                                    Object {
                                      "lineHeight": 24,
                                      "transform": Array [
                                        Object {
                                          "scaleX": 1,
                                        },
                                      ],
                                    },
                                    Object {
                                      "backgroundColor": "transparent",
                                    },
                                  ],
                                  Object {
                                    "fontFamily": "Material Design Icons",
                                    "fontStyle": "normal",
                                    "fontWeight": "normal",
                                  },
                                  Object {},
                                ]
                              }
                            >
                              Û∞çú
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              focusable={false}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              pointerEvents="box-none"
              style={
                Array [
                  Object {
                    "flex": 1,
                    "paddingHorizontal": 12,
                  },
                  Object {
                    "paddingHorizontal": 0,
                  },
                  Array [
                    Object {
                      "marginLeft": 12,
                    },
                    Object {
                      "alignItems": "center",
                      "bottom": 0,
                      "justifyContent": "center",
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    },
                    Array [
                      false,
                    ],
                  ],
                ]
              }
            >
              <Text
                accessibilityRole="header"
                accessible={true}
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "fontSize": 22,
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 28,
                    },
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontSize": 22,
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 28,
                      },
                      false,
                      undefined,
                    ],
                  ]
                }
              >
                Header title
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
